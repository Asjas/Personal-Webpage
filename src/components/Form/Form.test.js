import React from 'react';
import { render } from 'react-testing-library';
import ContactForm from './index';

describe('Contact form', () => {
  it('renders correctly', () => {
    const { container } = render(<ContactForm />);

    expect(container).toMatchSnapshot();
  });
});
