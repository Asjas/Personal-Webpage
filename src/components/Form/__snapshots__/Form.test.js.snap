// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact form renders correctly 1`] = `
.c0 {
  margin: 0 auto;
  padding: 25px 0;
  width: 500px;
  border-radius: 4px;
}

.c1 {
  display: grid;
  width: 100%;
  padding: 20px;
  border-radius: 6px;
}

.c1[disabled] {
  opacity: 0.5;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 1fr;
}

.c3 {
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  margin-bottom: 5px;
}

.c4 {
  border-radius: 4px;
  grid-column: 1/-1;
  grid-row: 2;
  height: 50px;
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  margin: 0 0 30px 0;
  padding: 0 0 0 20px;
  width: 100%;
}

.c4:focus {
  outline: none;
}

.c5 {
  border-radius: 4px;
  grid-column: 1/-1;
  grid-row: 2;
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  margin: 0 0 30px 0;
  min-height: 120px;
  outline: none;
  padding: 20px;
  resize: vertical;
  width: 100%;
}

.c6 {
  border-radius: 4px;
  color: #fff;
  cursor: pointer;
  outline: 0;
  padding: 14px;
  width: 100%;
}

<div>
  <form
    class="c0"
  >
    <fieldset
      aria-busy="false"
      class="c1"
      data-testid="form-disabled"
    >
      <label
        class="c2"
        for="name"
      >
        <span
          class="c3"
        >
          Enter your name:
        </span>
        <input
          class="c4"
          id="name"
          name="name"
          placeholder="full name"
          type="text"
          value=""
        />
      </label>
      <label
        class="c2"
        for="email"
      >
        <span
          class="c3"
        >
          Enter your email:
        </span>
        <input
          class="c4"
          id="email"
          name="email"
          placeholder="example@gmail.com"
          type="email"
          value=""
        />
      </label>
      <label
        class="c2"
        for="message"
      >
        <span
          class="c3"
        >
          Enter a message:
        </span>
        <textarea
          class="c5"
          id="message"
          name="message"
          placeholder="type your message here"
        />
      </label>
      <button
        class="c6"
        type="submit"
      >
        Submit
      </button>
    </fieldset>
  </form>
</div>
`;
